PERFORMANCE_CRITICAL_PATH.md

```diff
@@ -0,0 +1,10 @@
+# Performance Critical Path of user requests
+There're many files whose functions are in the critical path of read or write requests. They're so important to the overall performance that any regression will directly impact user experience. We add a comment #[PerformanceCriticalPath] to highlight these critical files. Please note that this is the best-effort work and some files in critical path may not be marked. But if a file is marked,  please pay special attention when you change its code.
+
+Here're some typical mistakes that should be avoided in the #[PerformanceCriticalPath] files.
+* Unnecessary synchronous I/O. Here 'unnecessary' means it's not a MUST for serving the current user request. For example, on_gc_snap() in peers.rs should not be in Fsm message handling.
```

Comment:
> on_gc_snap() in peers.rs should not be in Fsm message handling

This can be controversial. Some safety check cannot be done without accessing peer's field.

Reply:
refined the wording.